lib/src/client.dart

```diff
@@ -1753,8 +1752,8 @@ class Client extends MatrixApi {
         Logs().w('The user has been logged out!');
         await clear();
       }
-    } on IOException catch (e, s) {
-      Logs().w('Syncloop failed: Client has not connection to the server');
+    } on MatrixConnectionException catch (e, s) {
```

Comment:
This is the wrong exception here and never thrown by the sync loop

Reply:
Hmmm dow you have any other proposal ? Just a generic catch statement ?

Reply:
> Hmmm dow you have any other proposal

Nope. Needs some investigation which exception is thrown by http package but not depending on dart:io

Reply:
Should work ! Sorry, was in a meeting the entire time.